v0/destinations/drip/transform.js

```diff
@@ -0,0 +1,309 @@
+  if (!(id || email)) {
+    throw new CustomError("dripId or email is required for the call", 400);
+  }
+
+  let payload = constructPayload(message, identifyMapping);
+  payload.id = id;
+  payload.email = email;
+  payload.user_id = userId;
+
+  // id — >(yes) -> check if the user already exists (GET call): exists? continue: error;
+  // |
+  // (no)->email —>(yes) —> no need to check anything --> continue;
+  // 	|
+  // 	(no)—> visitor_uuid—>(yes) -> check if the visitor_uuid is present in the list: exists? continue: erorr;
+  // 				  |
+  // 				  (no)—> call can’t be made
+
+  if (!payload.first_name && !payload.last_name) {
+    const name = getFieldValueFromMessage(message, "name");
+    if (name && typeof name === "string") {
+      const [fname, lname] = name.trim().split(" ");
+      payload.first_name = fname;
+      payload.last_name = lname;
+    }
+  }
+  if (!payload.custom_fields) {
+    let customFields = {};
+    customFields = extractCustomFields(
+      message,
+      customFields,
+      ["traits", "context.traits"],
+      IDENTIFY_EXCLUSION_FIELDS
+    );
+    if (!isEmptyObject(customFields)) {
+      payload.custom_fields = customFields;
+    }
+  }
+
+  payload = removeUndefinedAndNullValues(payload);
+  const finalpayload = {
+    subscribers: [payload]
+  };
+  const basicAuth = Buffer.from(Config.apiKey).toString("base64");
+  const response = defaultRequestConfig();
+  response.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json"
+  };
+  response.method = defaultPostRequestConfig.requestMethod;
+  const campaignId =
+    getDestinationExternalID(message, "dripCampaignId") || Config.campaignId;
+  if (campaignId && email) {
+    const check = await createUpdateUser(finalpayload, Config, basicAuth);
+    if (!check) {
+      throw new CustomError("Unable to create/update user.", 400);
+    }
+
+    let campaignPayload = constructPayload(message, campaignMapping);
+    campaignPayload.email = email;
+
+    campaignPayload = removeUndefinedAndNullValues(campaignPayload);
+    const finalCampaignPayload = {
+      subscribers: [campaignPayload]
+    };
+
+    response.endpoint = `${ENDPOINT}/v2/${Config.accountId}/campaigns/${campaignId}/subscribers`;
+    response.body.JSON = finalCampaignPayload;
+    return response;
+  }
+  response.endpoint = `${ENDPOINT}/v2/${Config.accountId}/subscribers`;
+  response.body.JSON = finalpayload;
+  return response;
+};
+
+const trackResponseBuilder = async (message, { Config }) => {
+  const id = getDestinationExternalID(message, "dripId");
+
+  let email = getFieldValueFromMessage(message, "email");
+  if (!isValidEmail(email)) {
+    email = null;
+    logger.error("Enter correct email format.");
+  }
+  const event = getValueFromMessage(message, "event");
+  const ecomEvents = [
+    "order updated",
+    "order completed",
+    "order refunded",
+    "order cancelled",
+    "checkout started",
+    "fulfilled"
```

Comment:
lets do one thing. after lowecasing event name, for both `fulfilled` and `order fulfilled`, we should do the same thing